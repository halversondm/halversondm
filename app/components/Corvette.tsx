import * as React from "react";
import {PhotoGallery} from "./PhotoGallery";

export class Corvette extends React.Component<undefined, undefined> {

    render() {
        return (
            <div>
                <h2 className="text-primary">2001 Chevrolet Corvette Coupe</h2>
                <PhotoGallery perPage={9} totalPhotos={9} filePrefix="images/corvettePhotos/photo" fileSuffix=".jpg"/>
            </div>
        );
    }
}
