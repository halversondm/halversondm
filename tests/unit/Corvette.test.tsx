import React from "react";
import { render } from '@testing-library/react';
import Corvette from "../../app/components/Corvette";

test("Corvette component", () => {
    const unit = <Corvette />;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});