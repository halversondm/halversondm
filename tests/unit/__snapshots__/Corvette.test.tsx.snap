// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Corvette component 1`] = `"<div><div><h2 class="text-primary">2001 Chevrolet Corvette Coupe</h2><p>New to me in 2019. Purchased with 59,000 miles on the clock, Corsa Sport axle back exhaust, BBK cold air intake, short throw shifter and black logoed interior mats.</p><div id="photoGallery"><ul class="row photo-gallery"><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="0" class="img-fluid" src="images/corvettePhotos/photo1.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="1" class="img-fluid" src="images/corvettePhotos/photo2.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="2" class="img-fluid" src="images/corvettePhotos/photo3.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="3" class="img-fluid" src="images/corvettePhotos/photo4.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="4" class="img-fluid" src="images/corvettePhotos/photo5.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="5" class="img-fluid" src="images/corvettePhotos/photo6.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="6" class="img-fluid" src="images/corvettePhotos/photo7.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="7" class="img-fluid" src="images/corvettePhotos/photo8.jpg"></li><li class="col-md-2 col-lg-2 col-sm-3 col-xs-4"><img data-i="8" class="img-fluid" src="images/corvettePhotos/photo9.jpg"></li></ul><div class="row"><div><button type="button" data-id="1" class="btn btn-primary">Page 1</button></div></div></div></div></div>"`;
